src/PhpSpreadsheet/Reader/Gnumeric.php

```diff
@@ -77,16 +82,20 @@ public function canRead($pFilename)
         if (function_exists('gzread')) {
             // Read signature data (first 3 bytes)
             $fh = fopen($pFilename, 'rb');
-            $data = fread($fh, 2);
-            fclose($fh);
+            if ($fh !== false) {
+                $data = fread($fh, 2);
+                fclose($fh);
+            }
         }
 
         return $data == chr(0x1F) . chr(0x8B);
     }
 
-    private static function matchXml(string $name, string $field): bool
+    private static function matchXml(XMLReader $xml, string $expectedLocalName): bool
     {
-        return 1 === preg_match("/^(gnm|gmr):$field$/", $name);
+        return $xml->namespaceURI == self::NAMESPACE_GNM
```

Comment:
Should probably be using strict comparisons here, even if it means casting the xml node values explicitly to strings

Reply:
Done. I usually reflexively do this, but this particular section of code was copy-pasted without enough thought.

Reply:
I took the opportunity to change a few other comparisons to strict, when I was able to convince myself that was correct.